cli/src/terminal.rs

```diff
@@ -10,8 +12,98 @@ use std::{
 };
 use subtle_encoding::hex;
 use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, StandardStreamLock, WriteColor};
-use x509_parser::parse_x509_certificate;
-use yubikey::{certificate::Certificate, piv::*, YubiKey};
+use x501::attr::AttributeTypeAndValue;
+use x501::name::Name;
+use x509::Certificate;
+use yubikey::{certificate::read, piv::*, YubiKey};
+
+// the next three functions currently live in certval and may move to x509
+fn oid_lookup(oid: &ObjectIdentifier) -> String {
+    let oidstr = oid.to_string();
+    if oidstr == "2.5.4.41" {
+        return "name".to_string();
+    } else if oidstr == "2.5.4.4" {
+        return "sn".to_string();
+    } else if oidstr == "2.5.4.42" {
+        return "givenName".to_string();
+    } else if oidstr == "2.5.4.43" {
+        return "initials".to_string();
+    } else if oidstr == "2.5.4.44" {
+        return "generationQualifier".to_string();
+    } else if oidstr == "2.5.4.3" {
+        return "cn".to_string();
+    } else if oidstr == "2.5.4.7" {
+        return "l".to_string();
+    } else if oidstr == "2.5.4.8" {
+        return "st".to_string();
+    } else if oidstr == "2.5.4.9" {
+        return "street".to_string();
+    } else if oidstr == "2.5.4.11" {
+        return "ou".to_string();
+    } else if oidstr == "2.5.4.10" {
+        return "o".to_string();
+    } else if oidstr == "2.5.4.12" {
+        return "title".to_string();
+    } else if oidstr == "2.5.4.46" {
+        return "dnQualifier".to_string();
+    } else if oidstr == "2.5.4.6" {
+        return "c".to_string();
+    } else if oidstr == "2.5.4.5" {
+        return "serialNumber".to_string();
+    } else if oidstr == "2.5.4.65" {
+        return "pseudonym".to_string();
+    } else if oidstr == "0.9.2342.19200300.100.1.25" {
+        return "dc".to_string();
+    } else if oidstr == "1.2.840.113549.1.9.1" {
+        return "emailAddress".to_string();
+    }
```

Comment:
You should be able to make OID constants for each of these and compare those, and return a `&'static str`, something like:

```
const NAME_OID: ObjectIdentifier = ObjectIdentifier::new("2.5.4.41");

fn oid_lookup(oid: &ObjectIdentifier) -> &'static str {
    match oid {
        NAME_OID => "name",
        ...
    }
}

Reply:
OK. I will likely move these OIDs and lookup to x509. As noted in the comment, they are duplicated here from certval (but that's probably not the right spot for them either).

Reply:
As I was implementing this I recalled why String was used. It was to accommodate the case where the OID does not match and the dot notation form is returned.